tests/unit-tests/test_gsf.py

```diff
@@ -449,7 +450,45 @@ def test_create_builtin_lp_decoder():
     assert decoder.gamma == 6.
 
 
+def test_check_graph_name():
+    graph_name = "a"
+    check_graph_name(graph_name)
+    graph_name = "graph_name"
+    check_graph_name(graph_name)
+    graph_name = "graph-name"
+    check_graph_name(graph_name)
+    graph_name = "123-graph-name"
+    check_graph_name(graph_name)
+    graph_name = "_Graph-name"
+    check_graph_name(graph_name)
+
+    # test with invalid graph name
+    graph_name = "/graph_name"
+    invalid_name = False
+    try:
+        check_graph_name(graph_name)
+    except:
+        invalid_name = True
+    assert invalid_name
+
+    graph_name = "|graph_name"
+    invalid_name = False
+    try:
+        check_graph_name(graph_name)
+    except:
+        invalid_name = True
+    assert invalid_name
+
+    graph_name = "\graph_name"
+    invalid_name = False
+    try:
+        check_graph_name(graph_name)
+    except:
+        invalid_name = True
+    assert invalid_name
```

Comment:
Use `pytest.raises` to check for expected exceptions https://docs.pytest.org/en/7.1.x/reference/reference.html#pytest.raises